lenskit-core/src/main/java/org/lenskit/data/entities/Attribute.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * LensKit, an open source recommender systems toolkit.
+ * Copyright 2010-2014 LensKit Contributors.  See CONTRIBUTORS.md.
+ * Work on LensKit has been funded by the National Science Foundation under
+ * grants IIS 05-34939, 08-08692, 08-12148, and 10-17697.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package org.lenskit.data.entities;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Interner;
+import com.google.common.collect.Interners;
+import org.apache.commons.lang3.ClassUtils;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.grouplens.grapht.util.ClassProxy;
+
+import javax.annotation.Nonnull;
+import javax.annotation.concurrent.Immutable;
+import java.io.*;
+
+/**
+ * Identifier for an attribute in an entity.
```

Comment:
1. It seems strange for the class named `Attribute` to not actually represent an attribute, but rather the identifier of an attribute.
 2. The javadoc seems a little sparse. It says "what" the class is but it doesn't really explain the class. I don't think what I have here is right, but a proposal:

`This class represents the identifier for an attribute in an entity, and is used in accessing values stored in entities. Each entity is essentially a pair of a string name and a type. By coupling the name to its type we can ensure type safe access to the mixed type entity class`

Also, document the type parameter.